src/components/Admin/UserPage/DownloadCSV.tsx

```diff
@@ -0,0 +1,445 @@
+  const {isOpen, onOpen, onClose} = useDisclosure();
+  const [startYear, setStartYear] = useState(0);
+  const [startMonth, setStartMonth] = useState(0);
+  const [startDay, setStartDay] = useState(0);
+  const [endYear, setEndYear] = useState(0); // Change to timestamp
+  const [endMonth, setEndMonth] = useState(0); // Change to timestamp
+  const [endDay, setEndDay] = useState(0);
+  const [error, setError] = useState('');
+  /* ------------------------------------------------------------------- */
+
+  // Reset to starting state
+  function clearFields() {
+    setStartYear(0);
+    setStartMonth(0);
+    setStartDay(0);
+    setEndYear(0);
+    setEndMonth(0);
+    setEndDay(0);
+    setError('');
+  }
+
+  // Reset starting state and close modal
+  function handleClose() {
+    clearFields();
+    onClose();
+  }
+
+  /**
+   * Drop-down menu for months of the year
+   * @param value - state variable this input displays and changes
+   * @param setValue - function that sets the value
+   */
+  const MonthInput: ({value, setValue}: InputProps) => JSX.Element = ({
+    value,
+    setValue,
+  }: InputProps) => {
+    const labels = [
+      '',
+      'Jan',
+      'Feb',
+      'Mar',
+      'Apr',
+      'May',
+      'Jun',
+      'Jul',
+      'Aug',
+      'Sep',
+      'Oct',
+      'Nov',
+      'Dec',
+    ];
+    const options = [];
+    for (let i = 1; i < labels.length; i++) {
+      options.push(
+        <option value={i} key={i}>
+          {labels[i]}
+        </option>
+      );
+    }
+    return (
+      <Box width="30%">
+        <Select
+          type="number"
+          placeholder={t('downloadData.month')}
+          value={value}
+          onChange={event => {
+            setValue(+event.target.value);
+          }}
+        >
+          {options}
+        </Select>
+      </Box>
+    );
+  };
+
+  /**
+   * Drop-down menu for days of a month
+   * @param value - state variable this input displays and changes
+   * @param setValue - function that sets the value
+   */
+  const DayInput: ({value, setValue}: InputProps) => JSX.Element = ({
+    value,
+    setValue,
+  }: InputProps) => {
+    const options = [];
+    const maxDaysPerMonth = 31;
+    for (let i = 0; i < maxDaysPerMonth; i++) {
+      options.push(
+        <option value={i + 1} key={i}>
+          {i + 1}
+        </option>
+      );
+    }
+    return (
+      <Box width="30%">
+        <Select
+          placeholder={t('downloadData.day')}
+          size="md"
+          value={value}
+          onChange={event => {
+            setValue(+event.target.value);
+          }}
+        >
+          {options}
+        </Select>
+      </Box>
+    );
+  };
+
+  /**
+   * Check that input dates are valid, sets error accordingly
+   */
+  useEffect(() => {
+    const startDate = new Date(startYear, startMonth - 1, startDay);
+    const endDate = new Date(endYear, endMonth - 1, endDay);
+    if (startDate.getMonth() !== startMonth - 1) {
```

Comment:
Add comment explaining what kind of valid dates are checked here (i.e. you can't have a February 30th)